lib/elixir/pages/Writing Documentation.md

```diff
@@ -102,7 +102,13 @@ However keep in mind adding `@doc false` does not make the function private. The
 
   * Move the undocumented function to a module with `@moduledoc false`, like `MyApp.Hidden`, ensuring the function won't be accidentally exposed or imported. Remember you can use `@moduledoc false` to hide a whole module and still document each function with `@doc`. Tools will still ignore the module.
 
-  * Start the function name with underscores, for example, `__add__/2`, and add `@doc false`. The compiler does not import functions with leading underscores and they hint to anyone reading the code of their intended private usage.
+  * Start the function name with one or more underscores, for example, `__add__/2`, and add `@doc false`. The compiler does not import functions with leading underscores and they hint to anyone reading the code of their intended private usage.
+
```

Comment:
I'm new and it's my first time reading through this section. These recommendations are confusing to me because they seem to suggest a coupling between documentation being generated and a function being exposed to users.

As I understand it these decisions are decoupled (and should be). Authors can use `def` vs `defp` when deciding to expose the function to users, and use `@doc` to decide if they want to document the function. For example:

```elixir
defmodule MyApp.Example do
  @moduledoc """
  This is an example of a module with all 4 combinations of functions.
  """

  @doc "Documentation"
  def public_documented, do: 0

  @doc false
  def public_undocumented, do: 0

  @doc "Documentation"
  defp private_documented, do: 0

  @doc false
  defp private_undocumented, do: 0
end
```

Is that correct? If so I could take a shot at making the docs a little clearer. If I have something wrong, then I'm sorry to waste time! Like I said I'm very new at this.